scanner/hack/quay/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+)
+
+const (
+	findReposFmt = `https://quay.io/api/v1/find/repositories?page=%d&includeUsage=false`
+	listTagsFmt  = `https://quay.io/api/v1/repository/%s/%s/tag`
+)
+
+type RepoResponse struct {
+	Results []*Repository `json:"results"`
+}
+
+type Repository struct {
+	Namespace *Namespace `json:"namespace"`
+	Name      string     `json:"name"`
+}
+
+type Namespace struct {
+	Name string `json:"name"`
+}
+
+type TagResponse struct {
+	Tags []*Tag `json:"tags"`
+}
+
+type Tag struct {
+	Name string `json:"name"`
+}
+
+func fetchImages() []string {
+	fetchRepos := func(page int) []*Repository {
+		resp, err := http.Get(fmt.Sprintf(findReposFmt, page))
+		if err != nil {
+			panic(err)
+		}
+		defer func() {
+			_ = resp.Body.Close()
+		}()
+
+		var repoResp RepoResponse
+		if err := json.NewDecoder(resp.Body).Decode(&repoResp); err != nil {
+			panic(err)
+		}
+
+		return repoResp.Results
+	}
+
+	fetchTags := func(repo *Repository) []string {
+		resp, err := http.Get(fmt.Sprintf(listTagsFmt, repo.Namespace.Name, repo.Name))
+		if err != nil {
+			panic(err)
+		}
+		defer func() {
+			_ = resp.Body.Close()
+		}()
+
+		var tagsResp TagResponse
+		if err := json.NewDecoder(resp.Body).Decode(&tagsResp); err != nil {
+			panic(err)
+		}
+
+		tags := make([]string, 0, 10)
+		for i := 0; i < len(tagsResp.Tags) && i < cap(tags); i++ {
+			tag := tagsResp.Tags[i].Name
+			if strings.HasSuffix(tag, ".sig") || strings.HasSuffix(tag, ".sbom") {
+				continue
+			}
+			tags = append(tags, tag)
+		}
+
+		return tags
+	}
+
+	images := make([]string, 0, 2200)
+	for i := 0; i < 25; i++ {
```

Comment:
Super nit: The URL in `findReposFmt` looks to return `10` repos per page, and for each repo `10` tags are pulled (or less), and going through 25 pages of repos yields `2500` images (`25*10*10`) instead of `2200`.

Reply:
in theory, yes, but in practice, we got `2176`

Reply:
I can update it because we should allocate enough to hold the theoretical number of strings